manage_proposal.inc

```diff
@@ -0,0 +1,1141 @@
+        $cc = variable_get('research_migration_cc_emails', '');
+        $params['research_migration_proposal_approved']['proposal_id'] = $proposal_id;
+        $params['research_migration_proposal_approved']['user_id'] = $proposal_data->uid;
+        $params['research_migration_proposal_approved']['headers'] = array(
+            'From' => $from,
+            'MIME-Version' => '1.0',
+            'Content-Type' => 'text/plain; charset=UTF-8; format=flowed; delsp=yes',
+            'Content-Transfer-Encoding' => '8Bit',
+            'X-Mailer' => 'Drupal',
+            'Cc' => $cc,
+            'Bcc' => $bcc,
+        );
+        if (!drupal_mail('research_migration', 'research_migration_proposal_approved', $email_to, language_default(), $params, $from, true)) {
+            drupal_set_message('Error sending email message.', 'error');
+        }
+
+        drupal_set_message('esim research migration proposal No. ' . $proposal_id . ' approved. User has been notified of the approval.', 'status');
+        drupal_goto('research-migration-project/manage-proposal');
+        return;
+    } //$form_state['values']['approval'] == 1
+    else if ($form_state['values']['approval'] == 2) {
+        $query = "UPDATE {research_migration_proposal} SET approver_uid = :uid, approval_date = :date, approval_status = 2, dissapproval_reason = :dissapproval_reason WHERE id = :proposal_id";
+        $args = array(
+            ":uid" => $user->uid,
+            ":date" => time(),
+            ":dissapproval_reason" => $form_state['values']['message'],
+            ":proposal_id" => $proposal_id,
+        );
+        $result = db_query($query, $args);
+        /* sending email */
+        $user_data = user_load($proposal_data->uid);
+        $email_to = $user_data->mail;
+        $from = variable_get('research_migration_from_email', '');
+        $bcc = $user->mail . ', ' . variable_get('research_migration_emails', '');
+        $cc = variable_get('research_migration_cc_emails', '');
+        $params['research_migration_proposal_disapproved']['proposal_id'] = $proposal_id;
+        $params['research_migration_proposal_disapproved']['user_id'] = $proposal_data->uid;
+        $params['research_migration_proposal_disapproved']['headers'] = array(
+            'From' => $from,
+            'MIME-Version' => '1.0',
+            'Content-Type' => 'text/plain; charset=UTF-8; format=flowed; delsp=yes',
+            'Content-Transfer-Encoding' => '8Bit',
+            'X-Mailer' => 'Drupal',
+            'Cc' => $cc,
+            'Bcc' => $bcc,
+        );
+        if (!drupal_mail('research_migration', 'research_migration_proposal_disapproved', $email_to, language_default(), $params, $from, true)) {
+            drupal_set_message('Error sending email message.', 'error');
+        }
+
+        drupal_set_message('esim research migration proposal No. ' . $proposal_id . ' dis-approved. User has been notified of the dis-approval.', 'error');
```

Comment:
Rephrase as "eSim Research Migration with Proposal No. <propoosal_id> has been disapproved. The contributor has been notified of the disapproval."